inference-engine/src/plaidml_plugin/ops/embedding_bag_offsets_sum.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+using namespace plaidml::edsl;
+
+namespace {
+
+template <typename T>
+std::vector<T> cast_constant_operand(size_t operand_idx, ngraph::Node* layer) {
+  auto* ngraph_const = ngraph::as_type<ngraph::op::Constant>(layer->get_input_node_ptr(operand_idx));
+  if (ngraph_const) {
+    return ngraph_const->cast_vector<T>();
+  } else {
+    THROW_IE_EXCEPTION << "Dynamic slicing not currently supported by PlaidML plugin; all of indices, offsets and default index"
+                          "must be Constants.";
+  }
+}
+
+}  // namespace
+
+namespace PlaidMLPlugin {
+
+static OpRegistration reg("EmbeddingBagOffsetsSum", [](const Context& ctx) {
+  auto* layer = ngraph::as_type<ngraph::opset4::EmbeddingBagOffsetsSum>(ctx.layer);
+  IE_ASSERT(ctx.operands.size() >= 3);
+  IE_ASSERT(ctx.operands.size() <= 5);
+  auto I = ctx.operands.at(0);
+  auto indices = ctx.operands.at(1);
+  auto num_indices = cast_constant_operand<size_t>(1, layer).size();
+  auto offsets = cast_constant_operand<int32_t>(2, layer);
+  int32_t default_index = -1;
+  std::vector<float> per_sample_weights(num_indices, 1);
+
+  if (ctx.operands.size() >= 4) {
+    auto default_index_vec = cast_constant_operand<int32_t>(3, layer);
+    default_index = default_index_vec[0];
+    if (ctx.operands.size() == 5) {
+      per_sample_weights.clear();
+      per_sample_weights = cast_constant_operand<float>(4, layer);
+    }
+  }
+
+  auto batch = offsets.size();
+  offsets.push_back(num_indices);
+
+  auto I_gathered = gather(I, indices);
+
+  auto ndims = I_gathered.rank();
+  std::vector<TensorDim> I_dims(ndims);
+  std::vector<TensorIndex> I_idxs(ndims);
+  std::vector<Tensor> slices, Os;
+  I_gathered.bind_dims(I_dims);
+  auto O_dims = I_dims;
+  auto O_idxs = I_idxs;
+
+  O_dims[0] = edsl::TensorDim(1);
+  for (size_t i = 0; i < num_indices; ++i) {
+    O_idxs[0] = I_idxs[0] - i;
+    slices.push_back(edsl::Contraction(O_dims, O_idxs).sum(I_gathered(I_idxs)));
+  }
+
+  for (uint32_t l = 0; l < batch; ++l) {
+    if (offsets[l + 1] == offsets[l]) {
+      if (default_index == -1) {
+        auto zero = cast(Tensor(0.0), slices[0].dtype());
+        Os.push_back(select(slices[0] > 0.0, zero, zero));
```

Comment:
Ya this doesn't make any sense, both true and false cases of the select are zero, so the code is a no-op.

Reply:
Yes, we intend to use Os.push_back(slices[0] * 0) here.